lib/ProductOpener/DataQualityFood.pm

```diff
@@ -2696,6 +2696,67 @@ sub check_food_groups ($product_ref) {
 	return;
 }
 
+=encoding utf8
+
+=head2 check_incompatible_tags( PRODUCT_REF )
+
+Checks if 2 incompatible tags are assigned to the product
+
+To include more tags to this check, 
+add the property "incompatible:en" 
+at the end of code block in the taxonomy
+
+Example:
+en:Non-fair trade, Not fair trade
+fr:Non issu du commerce équitable
+incompatible_with:en: en:fair-trade
+
+=cut
+
+sub check_incompatible_tags ($product_ref) {
+
+	# list of tags having 'incompatible_with' properties
+	my @tags_to_check = ("categories", "labels");
+
+	foreach my $tag_to_check (@tags_to_check) {
+		$log->debug("check_incompatible_tags: tag_to_check $tag_to_check") if $log->debug();
+
+		my $incompatible_with_hash = get_all_tags_having_property($product_ref, $tag_to_check, "incompatible_with:en");
+
+		foreach my $key (keys %{$incompatible_with_hash}) {
+			my $value = %{$incompatible_with_hash}{$key};
+
+			$log->debug("check_incompatible_tags: key: " . $key . ", value: " . $value) if $log->debug();
+
+			# there can be more than a single value (comma (followed or not-followed by space (space is converted as hyphen) ) separated):
+			#   categories:en:short-grain-rices, categories:en:medium-grain-rices
+			my @all_incompatible_tags = split(/,-*/, $value);
+
+			foreach my $incompatible_tag (@all_incompatible_tags) {
+				$log->debug("check_incompatible_tags: incompatible_tag: " . $incompatible_tag) if $log->debug();
+
+				# split by ":" and produce 2 element list
+				#   for example, labels:en:contains-gluten -> (labels, en:contains-gluten)
+				my ($tagtype, $tagid) = split(/:/, $incompatible_tag, 2);
+
+				if (has_tag($product_ref, $tagtype, $tagid)) {
+					# column (:) prevents formating of the data quality facet on the website
+					$key =~ s/en://g;
+					$incompatible_tag =~ s/:en:/-/g;
+
+					# sort in alphabetical order to avoid facet a-b and facet b-a
+					my @incompatible_tags = sort ($tag_to_check . "-" . $key, $incompatible_tag);
+
+					add_tag($product_ref, "data_quality_errors",
+						"en:mutually-exclusive-$incompatible_tags[0]-and-$incompatible_tags[1]");
```

Comment:
Maybe mutually-exclusive-tags- as prefix ?

```suggestion
						"en:mutually-exclusive-tags-$incompatible_tags[0]-and-$incompatible_tags[1]");
```

Reply:
Shan't we also add a "en:mutually-exclusive-tags" tag, to be able to get all items with incompatibilities ?